### PR TITLE
Bump Keycloak version to 26.0.7

### PR DESCRIPTION
**Pull request by jedla97**
Bumping Keycloak to 26.0.5 and Keycloak client libs to 26.0.2. The introduction of `keycloak.client.version` is needed as Keycloak now have different release cycle for it's client libraries. You can read it [here](https://www.keycloak.org/docs/latest/release_notes/index.html#dedicated-release-cycle-for-the-client-libraries) 

Maybe this can be backported to 3.15 branch as this version of Keycloak is based on Quarkus 3.15. Don't know what are the preferences here.

Looking at [Keycloak release notes](https://www.keycloak.org/docs/latest/release_notes/index.html#keycloak-26-0-0) I don't see anything which should  be problematic. Only one think which can be problematic in future is [New generalized event types for credentials](https://www.keycloak.org/docs/latest/release_notes/index.html#new-generalized-event-types-for-credentials) (I found the `UPDATE_PASSWORD` in realm config file)

### CODE REVIEW

**Comment by sberyozkin** [comment]
Thanks @jedla97, I believe we have to wait until #43260 is merged...

**Comment by jedla97** [comment]
@sberyozkin thanks, didn't know about that. We can wait. I subscribe to that PR so I will update this when that PR is merged

**Comment by gastaldi** [comment]
It looks like `26.0.6` is already out. Would you mind upgrading to that version in this PR?

**Comment by jedla97** [comment]
> It looks like 26.0.7 is already out. Would you mind upgrading to that version in this PR?

This PR waiting for https://github.com/quarkusio/quarkus/pull/43260 to be merged, So after that is finished I'll update this to latest version

**Comment by jedla97** [comment]
The https://github.com/quarkusio/quarkus/pull/43260 is merged. I update the keycloak to the latest 26.0.7. I would like to see the test run to see if there are some related failiures so I can look at them

**Review by gastaldi — APPROVED** [review]

File: build-parent/pom.xml
Comment (sberyozkin) [inline build-parent/pom.xml:98]: @jedla97 Hi, I believe the client version should be aligned

Comment (jedla97) [reply build-parent/pom.xml:98]: The client libs have now dedicated release cycle https://www.keycloak.org/docs/26.0.0/release_notes/#dedicated-release-cycle-for-the-client-libraries so we would need to use `26.0.3` for keycloak itself as the client don't have 26.0.7 version (latest is 26.0.3). Which we using https://github.com/quarkusio/quarkus/blob/main/bom/application/pom.xml#L180

**Comment by sberyozkin** [comment]
@jedla97 Looks like all the OIDC tests fail. I won't really have much time now to investigate it right now, I'll have a look, and may have another look at 2nd or 3d Jan, and then on my return, the week starting 13th Jan. 
It may be that a single issue is causing problems, have a look at some single test in early Jan, see if you can spot something as well, thanks

**Comment by jedla97** [comment]
I'll look at it over the chrismas and new year holiday

**Comment by sberyozkin** [comment]
Thanks @jedla97, I'm also investigating, both of us will find something, enjoy the break

**Comment by sberyozkin** [comment]
@jedla97 Hi, it is only an HTML form change which is impacting the tests, so I updated it , there was also a oidc propagation exchange negative test message change.
Let's see how it goes, thanks

**Comment by sberyozkin** [comment]
@jedla97 All is green now,  thanks for initiating this PR

**Pull request merged**
